x-pack/filebeat/input/azureeventhub/v2_migration.go

```diff
@@ -0,0 +1,329 @@
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type consumerClient interface {
+	GetEventHubProperties(ctx context.Context, options *azeventhubs.GetEventHubPropertiesOptions) (azeventhubs.EventHubProperties, error)
+}
+
+type containerClient interface {
+	NewBlobClient(blobName string) *blob.Client
+	NewListBlobsFlatPager(o *container.ListBlobsFlatOptions) *runtime.Pager[container.ListBlobsFlatResponse]
+}
+
+type checkpointer interface {
+	SetCheckpoint(ctx context.Context, checkpoint azeventhubs.Checkpoint, options *azeventhubs.SetCheckpointOptions) error
+}
+
+// migrationAssistant assists the input in migrating
+// v1 checkpoint information to v2.
+type migrationAssistant struct {
+	log                 *logp.Logger
+	consumerClient      consumerClient
+	blobContainerClient containerClient
+	checkpointStore     checkpointer
+}
+
+// newMigrationAssistant creates a new migration assistant.
+func newMigrationAssistant(log *logp.Logger, consumerClient consumerClient, blobContainerClient containerClient, checkpointStore checkpointer) *migrationAssistant {
+	return &migrationAssistant{
+		log:                 log,
+		consumerClient:      consumerClient,
+		blobContainerClient: blobContainerClient,
+		checkpointStore:     checkpointStore,
+	}
+}
+
+// checkAndMigrate checks if the v1 checkpoint information for the partitions
+// exists and migrates it to v2 if it does.
+func (m *migrationAssistant) checkAndMigrate(ctx context.Context, eventHubConnectionString, eventHubName, consumerGroup string) error {
+	// Fetching event hub information
+	eventHubProperties, err := m.consumerClient.GetEventHubProperties(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get event hub properties: %w", err)
+	}
+
+	m.log.Infow(
+		"event hub information",
+		"name", eventHubProperties.Name,
+		"created_on", eventHubProperties.CreatedOn,
+		"partition_ids", eventHubProperties.PartitionIDs,
+	)
+
+	// Parse the connection string to get FQDN.
+	connectionStringInfo, err := parseConnectionString(eventHubConnectionString)
+	if err != nil {
+		return fmt.Errorf("failed to parse connection string: %w", err)
+	}
+
+	blobs, err := m.listBlobs(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, partitionID := range eventHubProperties.PartitionIDs {
+		err = m.checkAndMigratePartition(ctx, blobs, partitionID, connectionStringInfo.FullyQualifiedNamespace, eventHubName, consumerGroup)
+		if err != nil {
+			return fmt.Errorf("failed to check and migrate partition: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// checkAndMigratePartition checks if the v1 checkpoint information for the
+// `partitionID` partition.
+func (m *migrationAssistant) checkAndMigratePartition(
+	ctx context.Context,
+	blobs map[string]bool,
+	partitionID,
+	fullyQualifiedNamespace,
+	eventHubName,
+	consumerGroup string) error {
+
+	// v2 checkpoint information path
+	// mbranca-general.servicebus.windows.net/sdh4552/$Default/checkpoint/0
```

Comment:
Delete this line?

Reply:
Uhm, I need to rephrase this comment to make it more useful.